src/plugins/graph/adornments/movable-line/movable-line.tsx

```diff
@@ -292,29 +350,34 @@ export const MovableLine = observer(function MovableLine(props: IProps) {
           newIntercept = dragPivot.y - newSlope * dragPivot.x;
         }
 
-        lineObject.handleLower.classed('negative-slope', newSlope < 0);
-        lineObject.handleUpper.classed('negative-slope', newSlope < 0);
+        lineObject.handleLower?.classed('negative-slope', newSlope < 0);
+        lineObject.handleUpper?.classed('negative-slope', newSlope < 0);
 
-        model.dragLine(newIntercept, newSlope, instanceKey);
-        const lineModel = model.lines?.get(instanceKey);
+        model.dragLine(newIntercept, newSlope, lineObject.key);
+        const lineModel = model.lines.get(lineObject.key);
         if (lineSection === "lower") {
           lineModel!.setPivot1(dragPivot);
         } else {
           lineModel!.setPivot2(dragPivot);
         }
       }
-    }, [instanceKey, lineObject.handleLower, lineObject.handleUpper, model, xScaleCopy, yScaleCopy]),
+      refreshLines();
+    }, [model, refreshLines, xScaleCopy, yScaleCopy]),
```

Comment:
Shouldn't need to include `xScaleCopy` and `yScaleCopy` as dependents now that they're refs.